framework/QuartzRenderer/QuartzRenderer.swift

```diff
@@ -0,0 +1,599 @@
+                                       alpha: CGFloat(strokeColor.a)))
+        context.setLineWidth(CGFloat(thickness))
+        context.stroke(rect)
+    }
+
+    public func drawSolidRect(topLeftPoint p1: Point,
+                              topRightPoint p2: Point,
+                              bottomRightPoint p3: Point,
+                              bottomLeftPoint p4: Point,
+                              fillColor: Color = Color.white,
+                              hatchPattern: BarGraphSeriesOptions.Hatching,
+                              isOriginShifted: Bool) {
+        if (isOriginShifted) {
+            let w = abs(p2.x - p1.x)
+            let h = abs(p2.y - p3.y)
+            let y = min(p1.y,p2.y,p3.y,p4.y) + (0.1*plotDimensions.subHeight) + yOffset
+            let x = min(p1.x, p2.x, p3.x, p4.x) + xOffset + (0.1*plotDimensions.subWidth)
+            let rect = CGRect(x: Double(x),
+                              y: Double(y),
+                              width: Double(w),
+                              height: Double(h))
+            context.setFillColor(CGColor(red: CGFloat(fillColor.r),
+                                         green: CGFloat(fillColor.g),
+                                         blue: CGFloat(fillColor.b),
+                                         alpha: CGFloat(fillColor.a)))
+            context.fill(rect)
+            drawHatchingRect(x: x, y: y, width: w, height: h, hatchPattern: hatchPattern)
+        }
+        else {
+            let w: Float = abs(p2.x - p1.x)
+            let h: Float = abs(p2.y - p3.y)
+            let y = min(p1.y,p2.y,p3.y,p4.y) + yOffset
+            let x = p1.x + xOffset
+            let rect = CGRect(x: Double(x),
+                              y: Double(y),
+                              width: Double(w),
+                              height: Double(h))
+            context.setFillColor(CGColor(red: CGFloat(fillColor.r),
+                                         green: CGFloat(fillColor.g),
+                                         blue: CGFloat(fillColor.b),
+                                         alpha: CGFloat(fillColor.a)))
+            context.fill(rect)
+            drawHatchingRect(x: x, y: y, width: w, height: h, hatchPattern: hatchPattern)
+        }
+    }
+    
+    func drawHatchingRect(x: Float,
+                          y: Float,
+                          width w: Float,
+                          height h: Float,
+                          hatchPattern: BarGraphSeriesOptions.Hatching) {
+        switch (hatchPattern.rawValue) {
+        case 0:
+            break
+        case 1:
+            let drawPattern: CGPatternDrawPatternCallback = { _, context in
+                let line = CGMutablePath()
+                line.move(to: CGPoint(x: Double(0), y: Double(0)))
+                line.addLine(to: CGPoint(x: Double(10), y: Double(10)))
+                context.setStrokeColor(NSColor.black.cgColor)
```

Comment:
NSColor is a Mac-specific type and will cause problems when porting this to iOS. I'd recommend instead to use your platform-independent Color type and the extension I suggested earlier for returning a CGColor value from that.